### DISCUSSION TITLE
Clear R Studio server cache?

### DISCUSSION THREAD

**Discussion by yaaminiv** [RobertsLab/resources#1204] [Q&A]
## Is there a way to clear the R Studio server cache so it doesn't load any script/data that's causing it to freeze? 

A new entry in my R Studio server saga...

I am running [this R Markdown script](https://github.com/RobertsLab/project-oyster-oa/blob/master/code/Haws/04-methylKit.Rmd) on the R Studio server. I set up the Singularity session with [this script](https://github.com/RobertsLab/project-oyster-oa/blob/master/code/Haws/04-methylKit-RStudio.sh). I started the following code chunk yesterday afternoon:

```{r}
differentialMethylationStatsPloidy <- methylKit::calculateDiffMeth(methylationInformationFilteredCov5,
                                                                   covariates = covariatepH,
                                                                   overdispersion = "MN", test = "Chisq") #Calculate differential methylation statistics based on treatment indication from methRead. Include pH as a covariate. Use 40 cores.
head(differentialMethylationStatsPloidy) #Look at differential methylation statistics
```

I checked the R Studio server this morning and the chunk finished running! However, the R Studio server page itself was frozen, and I was unable to interact with the server at all. This is similar to something I mentioned with the emu and roadrunner R studio servers in [this discussion](https://github.com/RobertsLab/resources/discussions/1137#discussioncomment-449994). I closed my SSH tunnel, reopened it, and ran into the same issue. I also cancelled by SLURM script, ran it again, and opened a new tunnel, but I kept getting the following error message:

![Screen Shot 2021-05-08 at 1 54 33 PM](https://user-images.githubusercontent.com/22335838/117553327-bf8dd300-b005-11eb-9f7a-a2ed2d3964b6.png)

I've tried pressing "Wait", "Exit Page", and doing nothing. No matter what, I'm kicked out of the session:

![Screen Shot 2021-05-08 at 1 54 59 PM](https://user-images.githubusercontent.com/22335838/117553331-c3215a00-b005-11eb-9299-b377a1bb4add.png)

If I could open a new R Studio server session that didn't try loading my R Markdown script and packages when opening, I could load my saved data and keep moving forward (perhaps just by calling an R script in a SLURM script...). How do I do this?

**Comment by yaaminiv** [comment]
I refreshed the R Studio server browser page and closed out my script in 2 seconds and now it's not frozen.....I have no idea why that worked but it solves my problem in the meantime.

**Reply by yaaminiv** [reply]
Revised problem: 

I'm able to interact with the R Studio server until I open my R Markdown script. I have an [R script](https://github.com/RobertsLab/project-gigas-oa-meth/blob/master/code/06-methylKit.R) on the server as well, and do not encounter this issue with it! I will try the following:

- Deleting my R Markdown script and adding it again to `mox` with `rsync`
- Creating an R script version of my R Markdown script in case the problem is solely R Markdown

**Reply by yaaminiv** [reply]
> Deleting my R Markdown script and adding it again to mox with rsync

Got the "Page Unresponsive" error

![Screen Shot 2021-05-08 at 2 19 23 PM](https://user-images.githubusercontent.com/22335838/117553803-64111480-b008-11eb-9f7b-1e3bbceda99f.png)

**Reply by yaaminiv** [reply]
> Creating an R script version of my R Markdown script in case the problem is solely R Markdown

I created an [R script](https://github.com/RobertsLab/project-oyster-oa/blob/master/code/Haws/04-methylKit.R) and am able to interact with it on `mox`. My guess is that it's still something to do with the R Studio cache? But this is working for now.

**Comment by sr320** [comment]
In regular R Studio, in Settings you can tell it not to try re-opening prior scripts, data etc. when restarting.(ie starting fresh)
